client/pages/edit-trip.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import AppContext from '../lib/app-context';
+import formatDateEdit from '../lib/format-date-edit';
+
+export default class EditTrip extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: new Date().toISOString(),
+      endDate: new Date().toISOString(),
+      country: '',
+      budget: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/trips/${this.props.tripId}`)
+      .then(res => res.json())
+      .then(trip => {
+        this.setState(trip);
+      });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    fetch(`/api/trips/${this.props.tripId}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    })
+      .then(res => res.json());
+
+    this.setState({
+      startDate: new Date().toISOString(),
+      endDate: new Date().toISOString(),
+      country: '',
+      budget: ''
+    });
+
+    window.location.hash = '#trips';
```

Comment:
You can remove the `setState`, since we're transitioning away from this page anyway.

You should move your `window.location.hash` assignment into the `.then()` callback of your fetch so that you don't swap screens prematurely. In a live deployment, there will be a longer wait between the browser (client) and the server (somewhere in the midwest probably), so if you swap screens immediately, you may see a stale list because the req/res for updating the trip may not have completed yet.